patches/net/minecraft/network/protocol/handshake/ClientIntentionPacket.java.patch

```diff
@@ -19,18 +26,17 @@
  
     public ClientIntentionPacket(FriendlyByteBuf p_179801_) {
 -      this(p_179801_.readVarInt(), p_179801_.readUtf(255), p_179801_.readUnsignedShort(), ClientIntent.byId(p_179801_.readVarInt()));
-+      this(p_179801_.readVarInt(), p_179801_.readUtf(255), p_179801_.readUnsignedShort(), ClientIntent.byId(p_179801_.readVarInt()), p_179801_.readUtf(255));
++      this(p_179801_.readVarInt(), p_179801_.readUtf(255), p_179801_.readUnsignedShort(), ClientIntent.byId(p_179801_.readVarInt()), null);
     }
  
     @Override
-@@ -25,6 +_,7 @@
-       p_134737_.writeUtf(this.hostName);
+    public void write(FriendlyByteBuf p_134737_) {
+       p_134737_.writeVarInt(this.protocolVersion);
+-      p_134737_.writeUtf(this.hostName);
++      p_134737_.writeUtf(this.hostName + "\0" + this.fmlVersion + "\0");
```

Comment:
Why does this have two null chars?

Reply:
This was taken almost directly from the previous implementation: https://github.com/neoforged/NeoForge/blob/55133ebdb40c8e83d8d816b5bdf06c0683c48116/patches/minecraft/net/minecraft/network/protocol/handshake/ClientIntentionPacket.java.patch#L26. You'd probably have to ask cpw what the intention was with the trailing null char.

Reply:
I believe it might have been to easier extract our block of data in case another server (such as some bukkit one) decided to append data and didn't bother with their own separator

Reply:
Sounds like a reasonable explanation. Would you still consider that relevant enough to keep the trailing null char?